### ISSUE TITLE
Routing question: placeholders vs slash

### ISSUE THREAD

**Issue by skyflyer** [ThreeMammals/Ocelot#658]
## Expected Behavior

* Request to `/api/` would go to `/` downstream
* Request to `/api/test` would go to `/test` downstream

## Actual Behavior / Motivation for New Feature

* Reqeust to `/api/` is not resolved (see error below)
* Request to `/api/test` goes to downstream `/test

## Steps to Reproduce the Problem

  1. Have something listening on localhost:8000
  1. Configure ocelot to listen on localhost:5000
  1. Use config below

Config:
```json
{
    "ReRoutes": [
        {
            "DownstreamPathTemplate": "/{everything}",
            "DownstreamScheme": "http",
            "DownstreamHostAndPorts": [
                {
                    "Host": "localhost",
                    "Port": 8000
                }
            ],
            "UpstreamPathTemplate": "/api/{everything}",
            "UpstreamHttpMethod": [
                "GET",
                "POST"
            ]
        }
    ],
    "GlobalConfiguration": {
        "BaseUrl": "http://localhost:5000"
    }
}
```

## Specifications

  - Version: 12.0.1
  - Platform: mac
  - Subsystem: ?

## Comments

It is interesting to note, that if I configure ReRoute as `/{everything}` -> `/{everything}`, then it works, so it seems to me that `{everything}` wildcard match is not consistent in theese two cases.

**Comment by TomPallister** [comment]
@skyflyer 

```Request to /api/ would go to / downstream```

This doesn't work because /{everything} expects something after the slash, you need a reroute that is just /

```Request to /api/test goes to downstream `/test```

Not sure if this is a typo? I assume it's working?

**Comment by skyflyer** [comment]
So here is what is puzzling me (it is unexpected).

### Scenario 1

<details>
<summary>Config 1</summary>

```json
{
    "ReRoutes": [
        {
            "DownstreamPathTemplate": "/{everything}",
            "DownstreamScheme": "http",
            "DownstreamHostAndPorts": [
                {
                    "Host": "localhost",
                    "Port": 8000
                }
            ],
            "UpstreamPathTemplate": "/{everything}",
            "UpstreamHttpMethod": [
                "GET",
                "POST"
            ],
            "RateLimitOptions": {
                "EnableRateLimiting": true,
                "ClientWhiteList": [],
                "Period": "1s",
                "Limit": 1,
                "PeriodTimespan": 5
            }
        }
    ],
    "GlobalConfiguration": {
        "BaseUrl": "http://localhost:5000",
        "RateLimitOptions": {
            "DisableRateLimitHeaders": false,
            "QuotaExceededMessage": "Over limit",
            "HttpStatusCode": 429
          }
    }
}
```
</details>

Config1, where I configure that everything coming to ocelot, goes to `localhost:8000` works. So here, the pattern is `/{everything}` and it matches even `/` and also `/ocelot.json` and all works as one would expect.

* request to http://localhost:5000/ returns content from http://localhost:8000/
* request to http://localhost:5000/ocelot.json returns content from http://localhost:8000/ocelot.json

### Scenario 2

<details>
<summary>Config2</summary>

```json
{
    "ReRoutes": [
        {
            "DownstreamPathTemplate": "/{everything}",
            "DownstreamScheme": "http",
            "DownstreamHostAndPorts": [
                {
                    "Host": "localhost",
                    "Port": 8000
                }
            ],
            "UpstreamPathTemplate": "/api/{everything}",
            "UpstreamHttpMethod": [
                "GET",
                "POST"
            ],
            "RateLimitOptions": {
                "EnableRateLimiting": true,
                "ClientWhiteList": [],
                "Period": "1s",
                "Limit": 1,
                "PeriodTimespan": 5
            }
        }
    ],
    "GlobalConfiguration": {
        "BaseUrl": "http://localhost:5000",
        "RateLimitOptions": {
            "DisableRateLimitHeaders": false,
            "QuotaExceededMessage": "Over limit",
            "HttpStatusCode": 429
          }
    }
}
```
</details>

Based on the Config1, I would expect similar behaviour, only it is not, and I believe it should be (at least based on my experience with Apache and Nginx proxy setttings.

* request to http://localhost:5000/api return 404 (?)
* request to http://localhost:5000/api/ returns 404 (?)
* request to http://localhost:5000/api/ocelot.json returns content from http://localhost:8000/ocelot.json (expected)

Are you saying that if I want to achieve that, I would need to configure two `ReRoute`s? One for `/api/` and one for `/api/{everything}`?

**Comment by TomPallister** [comment]
@skyflyer yeh i think I might be wrong. This probably should work the way you expect but I don't know why it doesnt :/ I will mark this as a bug and see if we can get a fix!

**Comment by raman-m** [comment]
The issue has been [**accepted**](#event-10015522093) due to open PR #1333 by @AndyConlisk : 
- #1333

**Comment by raman-m** [comment]
Duplicate of #748

**Comment by raman-m** [comment]
### Fixed by #1911
